packages/cohere/src/cohere-chat-language-model.test.ts

```diff
@@ -760,33 +771,73 @@ describe('doGenerate', () => {
       expect(requestBody.documents).toBeUndefined();
     });
   });
+
+  it('should extract reasoning from response', async () => {
+    prepareJsonResponse({
+      text: '42',
+      thinking: 'So I was thinking ...',
+    });
+
+    const { content } = await model.doGenerate({
+      prompt: TEST_PROMPT,
+    });
+
+    expect(content).toMatchInlineSnapshot(`
+      [
+        {
+          "text": "42",
+          "type": "text",
+        },
+        {
+          "text": "So I was thinking ...",
+          "type": "reasoning",
+        },
+      ]
+    `);
+  });
 });
 
 describe('doStream', () => {
   function prepareStreamResponse({
-    content,
+    textContent,
+    thinkingContent = [],
     usage = {
       input_tokens: 17,
       output_tokens: 244,
     },
     finish_reason = 'COMPLETE',
     headers,
   }: {
-    content: string[];
+    textContent: string[];
+    thinkingContent?: string[];
     usage?: {
       input_tokens: number;
       output_tokens: number;
     };
     finish_reason?: string;
     headers?: Record<string, string>;
   }) {
+    const thinkingEvents = [];
+
+    if (thinkingContent.length > 0) {
+      thinkingEvents.push(
+        `event: content-start\ndata: {"type":"content-start","index":0,"delta":{"message":{"content":{"type":"thinking","thinking":""}}}}\n\n`,
```

Comment:
verify in original raw chunks that `index` is set to 0 for both thinking and text content